assignments/array-methods.js

```diff
@@ -2,82 +2,526 @@
+    shirt_size: "S",
+    company_name: "Quaxo",
+    donation: 55
+  },
+  {
+    id: 48,
+    first_name: "Anderea",
+    last_name: "MacGiolla Pheadair",
+    email: "amacgiollapheadair1b@xing.com",
+    shirt_size: "2XL",
+    company_name: "Kwimbee",
+    donation: 214
+  },
+  {
+    id: 49,
+    first_name: "Bel",
+    last_name: "Alway",
+    email: "balway1c@ow.ly",
+    shirt_size: "S",
+    company_name: "Voolia",
+    donation: 107
+  },
+  {
+    id: 50,
+    first_name: "Shell",
+    last_name: "Baine",
+    email: "sbaine1d@intel.com",
+    shirt_size: "M",
+    company_name: "Gabtype",
+    donation: 171
+  }
+];
 
 // ==== Challenge 1: Use .forEach() ====
-// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+// The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName.
 let fullName = [];
+
+runners.forEach(function(person) {
+  let name = `${person.first_name} ${person.last_name}`;
+  fullName.push(name);
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
 let allCaps = [];
-console.log(allCaps); 
+
+runners.map(function(firstName) {
+  let name = firstName.first_name.toUpperCase();
+  allCaps.push(name);
+});
+console.log(allCaps);
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
 let largeShirts = [];
+
+runners.filter(function(large) {
+  if (large.shirt_size === "L") {
+    let people = `${large.first_name} ${large.last_name}`;
+    largeShirts.push(people);
+  }
+});
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
 let ticketPriceTotal = [];
+
+ticketPriceTotal = runners.reduce(function(total, runner) {
+  return (total += runner.donation);
+}, 0);
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+//Get All runners email addresses.
+let email = [];
+runners.filter(function(runner) {
+  item = runner.email;
+  email.push(item);
+});
```

Comment:
Filter is used when you want an array of items from the original array that match some requirement. If all you want is just a portion of info from each object, use a map.
```suggestion
let email = runners.map(function(runner) {
return runner.email
})
});
```